HeterogeneousCore/CUDAUtilities/src/CachingDeviceAllocator.h

```diff
@@ -331,8 +331,7 @@ namespace notcub {
       cudaError_t error = cudaSuccess;
 
       if (device == INVALID_DEVICE_ORDINAL) {
-        if (CubDebug(error = cudaGetDevice(&entrypoint_device)))
-          return error;
+        cudaCheck(error = cudaGetDevice(&entrypoint_device));
```

Comment:
If all errors are handled with `cudaCheck()` internally, I'd expect there is no need to return an "error code" anymore (that was already passed through `cudaCheck()` in the calling code). Something to be cleaned up in a subsequent PR though.

Reply:
Since we don't expect errors, I would just `cudaCheck` all function calls (except for retrying the out-of-memory case).
Can be followed up later.

Reply:
> Since we don't expect errors, I would just `cudaCheck` all function calls (except for retrying the out-of-memory case).
> Can be followed up later.

I agree.